docs/website/contents/for-developers/Whitepaper.md

```diff
@@ -0,0 +1,244 @@
+|:-----------------------------------------------|:------------------------------------------------------------------------------------|
+| Forging Loop (aka The Mint)                    | Extend the Cardano chain by minting new blocks                                      |
+| `Mempool`                                      | Maintain a list of valid transactions to include in new blocks                      |
+| `ChainSync` and `BlockFetch` clients           | Gather candidate chains from peers                                                  |
+| `ChainSync` and `BlockFetch` servers           | Provide peers with access to `ChainDB` (see Storage Layer)                          |
+| `ChaineSel`                                    | Among all forks that can be fetched from peers, must identify the best chain.       |
+| N2N `TxSubmission`, N2C `TxSubmission` servers | Accept transactions from peers and store them into `Mempool` for further processing |
+|                                                |                                                                                     |
+|                                                |                                                                                     |
+|                                                |                                                                                     |
+
+## Responsibilities of the Storage Layer
+
+The Consensus layer needs to keep track of a significant amount of data to maintain the Cardano network. Some of this data is "hot" and relates to the candidate chains and other data is cold and relates to the historical chain.
+
+```mermaid
+flowchart TD
+    A("Storage") -- Sequential Access --> B("Immutable Chain")
+    A("Storage") -- Random Access --> C("Volatile Chain")
+    A("Storage") -- Random Access --> D("Recent Leder States")
+
+    subgraph noteB ["Efficient Rollback"]
+        C
+        D
+    end
+
+```
+
+In order to switch to an alternative chain, Consensus needs to evaluate the validity of such a chain. In order to evaluate the validity of a block, we need to have a Ledger state at the predecessor block. As applying blocks is not an inversible operation, this is usually solved by maintaining the last `k` ledger states in memory.
+
+In the Haskell reference implementation, the Storage layer is implemented by the `ChainDB` components, which comprises the `ImmutableDB`, `VolatileDB` and `LedgerDB` subcomponents:
+
+| Component   | Responsibility                                                                                            |
+|:------------|:----------------------------------------------------------------------------------------------------------|
+| ImmutableDB | Store definitive blocks on disk and provide iterator access to them for new syncing peers.                |
+| VolatileDB  | Store non-definitive blocks on disk, provide them to peers (random access?),                              |
```

Comment:
The starting point of a stream is random, then from that point onwards I think it is sequential. This might have changed with Genesis

Reply:
(This means that I agree with saying this is `random access`)

Reply:
I'll put a TODO to refine the requirements on access patterns for Immutable and Volatile databases so that we can discuss that further!